src/dialects/oracle/index.js

```diff
@@ -179,16 +179,22 @@ assign(Client_Oracle.prototype, {
       default:
         return response;
     }
-  }
+  },
 
+  // If the error is any of these, we'll assume we need to
+  // mark the connection as failed
+  isConnectionError(err) {
+    return [
+      'ORA-03114', // not connected to ORACLE
+      'ORA-03113', // end-of-file on communication channel
+      'ORA-03135', // connection lost contact
+      'ORA-12514', // listener does not currently know of service requested in connect descriptor
+      'NJS-040',
+      'NJS-024',
+      'NJS-003',
+      'NJS-024',
```

Comment:
NJS-024 is duplicated a few lines above.  And arguably shouldn't be on the list.